pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -218,14 +228,35 @@ func filterSeedsMatchingSeedSelector(cloudProfile *gardencorev1beta1.CloudProfil
 	return matchingSeeds, nil
 }
 
-func getCandidates(shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed, strategy config.CandidateDeterminationStrategy) ([]*gardencorev1beta1.Seed, error) {
+func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile, shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed) ([]*gardencorev1beta1.Seed, error) {
+	var providers []string
+	if cloudProfile.Spec.SeedSelector != nil {
+		providers = cloudProfile.Spec.SeedSelector.Providers
+	}
+
+	var matchingSeeds []*gardencorev1beta1.Seed
+	for _, seed := range seedList {
+		if seed.DeletionTimestamp == nil && matchProvider(seed.Spec.Provider.Type, shoot.Spec.Provider.Type, providers) && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed) {
```

Comment:
`seed.DeletionTimestamp == nil && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed) `   is checked in three different places. 
See [here](https://github.com/gardener/gardener/blob/547bdffba5d0487d1ab453801613109094cabfa4/pkg/scheduler/controller/shoot/scheduler_control.go#L328-L327) and [here](https://github.com/gardener/gardener/blob/547bdffba5d0487d1ab453801613109094cabfa4/pkg/scheduler/controller/shoot/scheduler_control.go#L340). 
I think it makes sense to introduce another function that filters out seed with deletion timestamps, with a taint and seeds that are not ready. This way we do not have to make the same check over again.

Reply:
yepp, but this was the case even before my change